### DISCUSSION TITLE
how to speed up computing

### DISCUSSION THREAD

**Discussion by wllshr123** [NanoComp/meep#2453] [Q&A]
Greetings! As a new user of Meep, I'm hoping to seek help from the community. My server has two 'Inter(R) Xeon(R) Gold 6342 CPU 2.8Ghz' processors, and I am running an adjoint optimization code. I am wondering how to fully utilize both processors for computing, similar to parallel computing. What additional code do I need to add to achieve this? Alternatively, will the code automatically use all CPU resources during runtime? (The adjoint optimization code I'm running is the mode converter example in Meep, and the code can be found [here](https://github.com/NanoComp/photonics-opt-testbed/blob/main/waveguide_mode_converter/mode_converter_meep_opt.py).)

Thank you in advance for your help!

**Comment by smartalecH** [comment]
> What additional code do I need to add to achieve this?

No additional code. Just launch your program with `mpirun` or `mpiexec`. More details are found [here in the manual](https://meep.readthedocs.io/en/latest/Parallel_Meep/).

Note that the adjoint solver will automatically parallelize too. Nothing special needs to be done with adjoint optimization compared to a normal simulation wrt parallelization. Keep in mind, though, that your degrees of freedom (DOF) will live on every process (along with the optimizer). So be mindful of memory usage etc.
